Sources/Models/DetectorType.swift

```diff
@@ -24,19 +24,18 @@
 
 import Foundation
 
-public enum DetectorType {
+public enum DetectorType: Hashable {
 
     case address
     case date
     case phoneNumber
     case url
     case transitInformation
+    case custom(regex: NSRegularExpression)
 
-    // MARK: - Not supported yet
-
-    //case mention
-    //case hashtag
-    //case custom
+    // swiftlint:disable force_try
+    public static var hashtag = DetectorType.custom(regex: try! NSRegularExpression(pattern: "#[a-zA-Z0-9]+", options: []))
+    public static var mention = DetectorType.custom(regex: try! NSRegularExpression(pattern: "@[a-zA-Z0-9]+", options: []))
```

Comment:
Do you think we should make these have a minimum of 4 characters to be detected?

Reply:
It can be a nice to do that but I do not know the standard in mention and hashtag. I can add it ? 
And if users want something more customisable they can add their own mention and hashtag

Reply:
@SD10 @JulienKode 4 characters seems arbitrary. what's your thinking @SD10?

Reply:
I'm behind the times. I guess 4 characters is arbitrary lol

Reply:
My opinion would be to adopt something closer to twitter's convention for hashtag because that is what people are used to.

**Cases not handled by the current regex**
#1234 pure numbers can be tags which can often be extensions
#MIT 4 is a pretty arbitrary requirement for the tag length. esp when people tag acryonyms
#swift_ios _ is a pretty common way to delimit words in a tag
discord#chat_room_number  discord uses # inside of their handle to share profiles
#hello#world : Matter of taste but twitter doesn't register this as a tag.

_Suggested Regex:_ "\B#\w*[a-zA-Z]+\w*"

The above regex handles the edge cases listed above except for the last one. It's not possible to implement without a look behind which is a little more complex of a regex.